### ISSUE TITLE
Does the AI work for height 4?

### ISSUE THREAD

**Issue by RisingOrange** [bpachev/plato#2]
I searched for connect 4 3d AIs to use for a personal project and found this repository. Your code looks pretty good, but in my use case I would need the program to work with a board height of 4 not 5. Could you tell me if the program work well when you change the "#define HEIGHT" statement in board.h to this number?

Also technically I am not allowed to use your code when your repository has no license.  Would you mind adding one?

**Comment by bpachev** [comment]
Good questions -
  For the most part the code *should* work if the height is changed to 4. I
think there are a couple places in the board evaluation function that could
cause problems. I'd be happy to take a look at the code and fix those
things, or if you'd prefer to submit a pull request you're welcome to do
so.
As far as a license goes, I wrote this code for fun, so you're welcome to
use it. I'll add one to the repo.
  Let me know if you have more questions -
    Benjamin.

**Comment by RisingOrange** [comment]
Great! Thanks for the helpful response.

I was testing the code for a bit (using the default height of 5) and I think I found a test case for which the program should return a different move. 

I used tests.c with the testMovePick method (I provided "m" as the commandline argument for the mode).
This is are the contents of my_test.txt:
```
5 0 3
0 0
0 1
0 1
0 2
0 2
0 3
0 3
0 3
0 2
```

The output was:
```
Test script for Pillars of Plato AI.
Testing my_test.txt
White move: 0 0, stacknum 0
Black move: 0 1, stacknum 1
White move: 0 1, stacknum 1
Black move: 0 2, stacknum 2
White move: 0 2, stacknum 2
Black move: 0 3, stacknum 3
White move: 0 3, stacknum 3
Black move: 0 3, stacknum 3
White move: 0 2, stacknum 2
Expected 0 3, picked move 1 3!
Test case my_test.txt failed! 
```

Shouldn't Black prevent White's win on the diagonal here?

**Comment by bpachev** [comment]
Yes - this looks like a bug. I'll investigate and get back to you.

**Comment by bpachev** [comment]
Sorry for the longer turnaround time - this is actually not a bug. The test
position you are using has two threats to win for White - Black can't stop
both of them. Hence the choice of move doesn't change the outcome of the
game. The position after playing all the test moves is
 _ _ W B
 _WWW
 WBBB
White is threatening to win both with the move 0 0 and the move 0 3 - Black
can't stop it.
When the AI is in a position where multiple moves lead to the same outcome,
it picks a random move. This is so that it won't play the same game each
time.
Hope this helps - and thanks again for your interest in the code!

**Comment by bpachev** [comment]
OK - I went through the code and fixed everything that assumes the height
was 5 - it should all work with a height of 4 if you change
#define HEIGHT 5 to #define HEIGHT 4. I ran the test suite with the
height equal to 4, and everything seemed to pass except for the tests that
make use of the fifth level. Please let me know if you run into any issues.
  Happy coding!

Benjamin
